core/shared/src/main/scala/scalaz/zio/UIO.scala

```diff
@@ -0,0 +1,534 @@
+package scalaz.zio
+
+import scalaz.zio.Exit.Cause
+import scalaz.zio.internal.{ Executor, Platform }
+
+/*
+ * PHASE 1 DONE
+ * PHASE 2 DONE
+ * PHASE 3 DONE
+ * PHASE 4 DONE
+ */
+
+/*
+ * EXTENDS ZIOFunctions
+ */
+object UIO {
+
+  type UioE = Nothing
+  type UioR = Any
+
+  def apply[A](a: => A): UIO[A] = effectTotal(a)
+
+  /*
+   * ---------------------------------------------------------------------------------------
+   * The Following is taken from ZIOFunctions.
+   * ---------------------------------------------------------------------------------------
+   */
+
+  /** Returns an effect that models failure with the specified error.
+   * The moral equivalent of `throw` for pure code.
+   */
+  final def fail(error: UioE): IO[UioE, Nothing] = ZIO.fail(error)
```

Comment:
Replace all the `IO[UioE, ...]` with `UIO[...]`.

Reply:
Ah, we can also delete `fail` since the user cannot call the method (no values of type `Nothing` exist). 😄